docs/pktoken.md

```diff
@@ -0,0 +1,702 @@
+
+| Commitment Type               | Commitment Location  | Claim name |
+| -------------                 |--------------        |--------------|
+| Nonce-Commitment              | `payload`            | `nonce`      |
+| Audience-Commitment           | `payload`            | `aud`        |
+| GQ-Commitment                 | `OP-protected header`| `CIC`        |
+| ZK-Commitment                 | `OP-protected header`| `CIC`        |
+
+### Types of Signatures in a PK Token
+
+#### OP (OpenID Provider) Signature
+
+This is signature of the OpenID Provider that created the payload and signed it to create the ID Token. This signature is responsible for binding the identity claims in the payload to that identity's public key.
+
+Typically OP signature/protected header along with the payload is the ID Token. For some more advanced forms of PK Tokens such as GQ or ZK PK Tokens, we transform OP signature and protected header. In these cases, you can no longer recover the ID Token from the PK Token, but even when we alter the OP Signature and protected header in this way, we don't break the cryptographic binding between payload and OpenID Provider that the signature provides.
+
+Some OpenID Providers do not specify `typ`. To get around this we classify a signature as being from the OpenID Provider if either `typ=JWT` or `typ` is not defined.
+
+#### CIC (Client-Instance Claims) Signature
+
+The Client-Instance is the identity's OpenPubkey client. The CIC are the claims made about the identity by this client. The CIC signature performs two functions. First, it provides the needed data to allow verifiers to check that an ID Token has a binding to the user's public key by opening the commitment. Second, it functions as a [Proof-of-Possession](https://csrc.nist.gov/glossary/term/proof_of_possession) showing that the identity's knows the signing key associated with the public key `upk`. This Proof-of-Possession prevents rogue key attacks in which a attackers associates their identity with another identity's public key and the attempts to claim that they produced signatures produced by the other identity.
+
+The CIC always contains the following claims (and may contain other claims):
+
+* `alg` - the algorithm of both the signature and the identity's public key.
+* `upk` - the JWK (JSON Web Key) of the identity's public key. UPK stands for User's Public Key.
+* `typ` - this value is always set to `CIC` to identify this protected header and signature pair as the CIC.
+
+#### COS (Cosigner) Signature
+
+The Cosigner is a third party who issues a signature if they are able to authenticate the identity in the ID Token. This authentication must be independent of the OpenID Provider's authentication. The purpose of the Cosigner is enable OpenPubkey to maintain security even if the OpenID Provider becomes maliciously.
+
+Cosigning is an optional feature of OpenPubkey and OpenPubkey can be used without a cosigner. In such cases there is no COS signature.
```

Comment:
Can remove "and OpenPubkey can be used without a cosigner." as the first part of the sentence addresses that.